R/score.R

```diff
@@ -225,3 +223,58 @@ apply_rules <- function(data, metrics, ...) {
   }, data, ...)
   return(data)
 }
+
+
+#' Construct An Object Of Class `scores`
+#' @description This function creates an object of class `scores` based on a
+#' data.table or similar.
+#' @param scores A data.table or similar with scores as produced by [score()]
+#' @param score_names A character vector with the names of the scores
+#' (i.e. the names of the scoring rules used for scoring)
+#' @keywords internal
+#' @examples
+#' \dontrun{
+#' df <- data.frame(
+#'   model = "A",
+#'   wis = "0.1"
+#' )
+#' new_scores(df, "wis")
+#' }
+new_scores <- function(scores, score_names) {
+  scores <- as.data.table(scores)
+  class(scores) <- c("scores", class(scores))
+  setattr(scores, "score_names", score_names)
+  return(scores[])
+}
+
+
+#' Create An Object Of Class `scores` From Data
+#' @description This convenience function wraps [new_scores()] and validates
+#' the `scores` object.
+#' @inheritParams new_scores
+#' @returns Returns an object of class 1scores`
+#' @importFrom checkmate assert_data_frame
+#' @keywords internal
```

Comment:
we could export these to give users the option of creating their own but with some wording to say only if they know what they are doing but not required and perhaps can wait on it.